src/localFetch.ts

```diff
@@ -0,0 +1,42 @@
+import nodeFetch, { RequestInfo, RequestInit } from "node-fetch"
+
+import { IFetchComponent } from "@well-known-components/http-server"
+import { IConfigComponent } from "@well-known-components/interfaces"
+
+// start TCP port for listeners
+let lastUsedPort = 19000 + parseInt(process.env.JEST_WORKER_ID || "1") * 1000
+
+function getFreePort() {
+  return lastUsedPort + 1
+}
+
+/**
+ * Local Fetch component for testing local urls
+ * @public
+ */
+export async function createLocalFetchCompoment(
+  configComponent: IConfigComponent
+): Promise<IFetchComponent> {
+  const currentPort = getFreePort()
+
+  Object.assign(process.env, {
+    HTTP_SERVER_HOST: "0.0.0.0",
+    HTTP_SERVER_PORT: (currentPort + 1).toString(),
+  })
```

Comment:
This assumes that the selected `config` component uses `process.env`.

I'd return instead a composed `config` component with the brand new `localFetch` component in a way that the new config can replace (or compose) the provided config. Avoiding future incompatibilities


```ts

const injectedMap = {
  HTTP_SERVER_HOST: "0.0.0.0",
  HTTP_SERVER_PORT: (currentPort + 1).toString(),
}

const composedConfig = {
  async getString(key) {
    if (injectedMap.hasOwnProperty(key)) return injectedMap[key]
    return config.getString(key)
  },
  async requireString(key) {
    if (injectedMap.hasOwnProperty(key)) return injectedMap[key]
    return config.requireString(key)
  },
  async getNumber(key) {
    if (injectedMap.hasOwnProperty(key)) {
      const value = parseNumber(injectedMap[key])
      if (value != undefined && value !== null) {
        const numValue = parseFloat(value)

        if (isNaN(numValue)) {
          throw new Error(`Configuration: config "${name}" should be a number, got ${typeof value} (${value}) instead`)
        }

        return numValue
      }
    }
    return config.getNumber(key)
  },
  async requireNumber(key) {
    if (injectedMap.hasOwnProperty(key)) {
      const value = parseNumber(injectedMap[key])
      if (value != undefined && value !== null) {
        const numValue = parseFloat(value)

        if (isNaN(numValue)) {
          throw new Error(`Configuration: config "${name}" should be a number, got ${typeof value} (${value}) instead`)
        }

        return numValue
      }
    }
    return config.requireNumber(key)
  },
}

return { localFetch, config: composedConfig }

```

It may be a good idea to abstract that behavior to a `composeConfigComponent(...components)` in https://github.com/well-known-components/env-config-provider/blob/main/src/index.ts